systemtest/src/main/java/io/strimzi/systemtest/kafkaclients/KafkaClientProperties.java

```diff
@@ -0,0 +1,458 @@
+                    "required " +
+                    "oauth.client.id=\"" + clientId + "\" " +
+                    "oauth.client.secret=\"" + clientSecretName + "\" " +
+                    "oauth.token.endpoint.uri=\"" + oauthTokenEndpointUri + "\";");
+
+            return this;
+        }
+
+        public KafkaClientPropertiesBuilder withSaslJassConfigAndTls(String clientId, String clientSecretName, String oauthTokenEndpointUri) {
+
+            try {
+                importKeycloakCertificateToTruststore(properties);
+                fixBadlyImportedAuthzSettings();
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+
+            if (clientId.isEmpty() || clientSecretName.isEmpty() || oauthTokenEndpointUri.isEmpty()) {
+                throw new InvalidParameterException("You do not specify client-id, client-secret name or oauth-token-endpoint-uri inside kafka client!");
+            }
+
+            properties.setProperty(SaslConfigs.SASL_JAAS_CONFIG,
+                "org.apache.kafka.common.security.oauthbearer.OAuthBearerLoginModule " +
+                    "required " +
+                    "oauth.client.id=\"" + clientId + "\" " +
+                    "oauth.client.secret=\"" + clientSecretName + "\" " +
+                    "oauth.token.endpoint.uri=\"" + oauthTokenEndpointUri + "\" " +
+                    "oauth.ssl.endpoint.identification.algorithm=\"\"" +
+                    "oauth.ssl.truststore.location=\"" + properties.get(SslConfigs.SSL_TRUSTSTORE_LOCATION_CONFIG) + "\" " +
+                    "oauth.ssl.truststore.password=\"" + properties.get(SslConfigs.SSL_TRUSTSTORE_PASSWORD_CONFIG) + "\" " +
+                    "oauth.ssl.truststore.type=\"" + properties.get(SslConfigs.SSL_TRUSTSTORE_TYPE_CONFIG) + "\" ;");
+
+            return this;
+        }
+
+        /**
+         * Create properties which are same pro producer and consumer
+         */
+        public KafkaClientPropertiesBuilder withSharedProperties() {
+            // For turn off hostname verification
+            properties.setProperty(SslConfigs.SSL_ENDPOINT_IDENTIFICATION_ALGORITHM_CONFIG, "");
+
+            try {
+                if (!properties.getProperty(CommonClientConfigs.SECURITY_PROTOCOL_CONFIG).equals(CommonClientConfigs.DEFAULT_SECURITY_PROTOCOL) &&
+                    !properties.getProperty(CommonClientConfigs.SECURITY_PROTOCOL_CONFIG).equals("SASL_" + CommonClientConfigs.DEFAULT_SECURITY_PROTOCOL)
+                ) {
+                    Secret clusterCaCertSecret = kubeClient(namespaceName).getSecret(caSecretName);
+                    File tsFile = File.createTempFile(KafkaClientProperties.class.getName(), ".truststore");
+                    String tsPassword = "foo";
+                    KeyStore ts = KeyStore.getInstance(TRUSTSTORE_TYPE_CONFIG);
+                    if (caSecretName.contains("custom-certificate")) {
+                        tsFile.deleteOnExit();
```

Comment:
since we should always delete these it's better to remove from both branches and put it immediately after it's creation.